backend/api/post/postService.js

```diff
@@ -31,21 +32,21 @@ module.exports = function (app) {
   });
 
   app.get("/api/posts/:id", async (req, res) => {
-    let post;
     try {
-      data = await postModel
-        .find({ _id: req.params.id })
+      let post = await postModel
+        .findOne({ _id: req.params.id })
         .lean()
         .populate("categoryId")
         .populate("moderatorsIds", ["username"])
-        .populate("ownerId", ["username", "roles"]);
-      for (post of data) {
-        const comments = await commentModel.find({ postId: req.params.id });
-        const commentLength = comments.length;
-        post = { ...post, comments, commentLength };
-      }
+        .populate("ownerId", ["username", "roles"])
+        .exec();
+      
+      const comments = await commentModel.find({ postId: req.params.id });
+      const commentLength = comments.length;
+
+      post = { ...post, comments, commentLength };
       res.status(200).json(post);
-    } catch (e) {
+    } catch (error) {
       res.sendStatus(404);
```

Comment:
403 eller 404? märkte att det används olika på olika ställen

Reply:
Just denna gäller för routen /api/posts/:id så den är öppen för alla men kastar 404 (not found) status om id'et inte existerar.